README.md

```diff
@@ -502,6 +503,43 @@ class MyGem::Inflector < Zeitwerk::GemInflector
 end
 ```
 
+<a id="markdown-the-on_load-callback" name="the-on_load-callback"></a>
+### The `on_load` callback
+
+The usual place to run something when a file is loaded is the file itself. However, sometimes you'd like to be called, and this is possible with the `on_load` callback.
+
+For example, let's say a class managed by Zeitwerk in a web application has a class-level attribute whose value depends on the environment:
+
+```ruby
+class SomeApiClient
+  class << self
+    attr_accessor :endpoint
+  end
+end
+```
+
+With `on_load`, it is easy to initialize said attribute at boot time according to the configuration:
+
+```ruby
+# config/environments/development.rb
+loader.on_load("SomeApiClient") do
+  SomeApiClient.endpoint = "https://api.dev"
+end
+
+# config/environments/production.rb
+config.on_load("SomeApiClient") do
+  SomeApiClient.endpoint = 'https://api.prod'
+end
+```
+
+This method receives the name of the target class or module as a string. The given block is executed every time its corresponding file is loaded. That includes reloads.
+
+Multiple callbacks on the same target are supported, and they run in order of definition.
+
+Think about `on_load` as reacting to an event. In particular, if the target was already loaded when the callback is defined, the block won't run. Normally, these callbacks are set before `setup`, but this is not enforced.
+
+Defining a callback for a target not managed by the receiver is not an error, the block simply won't be ever executed.
```

Comment:
Not a native speaker but I think "won't ever be executed" is more idiomatic.

Reply:
Confirmed that yours is the good one with a native speaker 👍.